test/tools/yulopti.cpp

```diff
@@ -171,28 +208,118 @@ class YulOpti
 		}
 	}
 
+	void objectApply(shared_ptr<Object> _object, function<void(Object&)> _fn)
+	{
+		for (auto const& subObjectNode: _object->subObjects) {
+			auto subObject = dynamic_pointer_cast<Object>(subObjectNode);
+
+			if (subObject != nullptr)
+				objectApply(subObject, _fn);
+		}
+
+		_fn(*_object);
+	}
+
+	void analyze(ErrorReporter& errorReporter)
+	{
+		objectApply(m_object, [&](Object& object) {
+			object.analysisInfo = make_shared<yul::AsmAnalysisInfo>();
+
+			AsmAnalyzer analyzer(
+				*object.analysisInfo,
+				errorReporter,
+				m_dialect,
+				{},
+				object.qualifiedDataNames()
+			);
+
+			bool success = analyzer.analyze(*object.code);
+			yulAssert(success && !errorReporter.hasErrors(), "Invalid assembly/yul code.");
+		});
+	}
+
 	void disambiguate()
 	{
-		*m_ast = std::get<yul::Block>(Disambiguator(m_dialect, *m_analysisInfo)(*m_ast));
-		m_analysisInfo.reset();
-		m_nameDispenser.reset(*m_ast);
+		objectApply(m_object, [&](Object& object) {
```

Comment:
Same as before regarding breaking lines, capture and coding style. 
Please make sure to also check the next set of functions that have this lambda as a parameter of `objectApply`.

Reply:
Fixed